### PR TITLE
Updating 4.3-dev with changes from 4.2-dev

### PR DESCRIPTION
**Pull request by obuisard**

### CODE REVIEW

**Comment by obuisard** [comment]
I have spent hours fixing conflicts in package-lock.json.
I did it to the best of my ability in trying to understand the changes, what needed to be changed and what needed to remain in 4.3. 
However, I would really appreciate someone checking the file out to ensure I did not make any mistake. There is no way there are no mistakes in the file over the >50 conflicts that needed to be fixed.

Thank you for your help.
@HLeithner @wilsonge

**Comment by wilsonge** [comment]
Drone failed because package-lock isn't representing what's in the package.json. My best advice is don't care. This branch isn't out yet and whilst it's not ideal to keep bumping dependencies through beta - the easiest way to fix this is to just run an npm i and grab the latest dependencies and be done with it. It's nearly impossible to keep those in sync like you say...

**Comment by HLeithner** [comment]
as george said, you have to ignore the package-lock.json and the composer.lock just rebuild it if they conflict.

**Comment by obuisard** [comment]
Thank you all for your help and advice. Traveling right now so I will hold this until Friday for such procedure that requires full attention and processing power :-)

File: .drone.yml [hunk 1]
```diff
@@ -179,143 +159,174 @@ steps:
...
     image: joomlaprojects/docker-images:systemtests
-    volumes:
-      - name: cypress-cache
-        path: /root/.cache/Cypress
     commands:
-      - sed -i 's/tests\\/Codeception\\/_output/\\/drone\\/src\\/tests\\/cypress\\/output/' codeception.yml
+      - sed -i 's/tests\\/Codeception\\/_output/\\/drone\\/src\\/tests\\/Codeception\\/_output/' codeception.yml
       - php libraries/vendor/bin/codecept build
-      - npx cypress install
-      - npx cypress verify
```
Comment (heelc29) [inline .drone.yml]: The changes in `.drone.yml` don't look good (e.g. remove cypress)

**Comment by brianteeman** [comment]
Maybe this was the wrong way to do it but this is what I did and it failed :(

1. apply PR
2. delete the two lcok files
3. comppser install
4. npm install

It failed on step 4 with the following error which I will happily admit to failing to understand
```

179 verbose stack Error: command failed
179 verbose stack     at ChildProcess.<anonymous> (C:\Users\brian\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\promise-spawn\lib\index.js:63:27)
179 verbose stack     at ChildProcess.emit (node:events:513:28)
179 verbose stack     at maybeClose (node:internal/child_process:1098:16)
179 verbose stack     at ChildProcess._handle.onexit (node:internal/child_process:304:5)
180 verbose pkgid joomla@4.0.0
181 verbose cwd C:\htdocs\j42
182 verbose Windows_NT 10.0.22623
183 verbose node v19.1.0
184 verbose npm  v8.19.2
185 error code 1
186 error path C:\htdocs\j42
187 error command failed
188 error command C:\WINDOWS\system32\cmd.exe /d /s /c node build/build.js --prepare
189 verbose exit 1
190 timing npm Completed in 114090ms
191 verbose code 1
```

**Comment by brianteeman** [comment]
I see now that drone reported the exact same error

**Comment by dgrammatiko** [comment]
@obuisard @brianteeman my bad I guess, update the scripts part of the package.json to:

```json
    "build:js": "node build/build.js --compile-js",
    "build:css": "node build/build.js --compile-css",
    "build:bs5": "node build/build.js --compile-bs",
    "build:com_media": "NODE_ENV=PRODUCTION node build/build.js --com-media",
    "build:com_media:dev": "NODE_ENV=DEVELOPMENT node build/build.js --com-media",
    "watch": "node build/build.js --watch",
    "watch:com_media": "NODE_ENV=DEVELOPMENT node build/build.js --watch-com-media",
    "lint:js": "eslint --config build/.eslintrc --ignore-pattern '/media/' --ext .es6.js,.es6,.vue .",
    "lint:css": "stylelint --config build/.stylelintrc.json \"administrator/components/com_media/resources/**/*.scss\" \"administrator/templates/**/*.scss\" \"build/media_source/**/*.scss\" \"templates/**/*.scss\" \"installation/template/**/*.scss\"",
    "install": "NODE_ENV=PRODUCTION node build/build.js --prepare",
    "update": "NODE_ENV=PRODUCTION node build/build.js --copy-assets && node build/build.js --build-pages && node build/build.js --compile-js && node build/build.js --compile-css && node build/build.js --compile-bs && node build/build.js --com-media",
    "gzip": "node build/build.js --gzip",
    "versioning": "node build/build.js --versioning",
    "browserlist:update": "npx browserslist@latest --update-db",
    "cypress:install": "cypress install",
    "cypress:open": "cypress open",
    "cypress:run": "cypress run"
```

There are couple of missing env variables...

**Comment by brianteeman** [comment]
dam - just came back to report that it was all your fault as there were missing variables

/me wonders if anyojne tested the pr

**Comment by brianteeman** [comment]
@dgrammatiko your change didnt work. it seems to be a windows issue that needs different sytax when using node_env than you have used.

https://stackoverflow.com/questions/9249830/how-can-i-set-node-env-production-on-windows

**Comment by dgrammatiko** [comment]
> /me wonders if anyojne tested the pr

They did and actually the PR without the extra ENV vars is working perfectly on the 4.2 repo. So basically this is the best proof that ranges both in package.json and composer.json are **extremely dangerous** as you end up with versions that might actually break the system. I keep repeating, that both these sons should point to **SPECIFIC** versions of the packages but I guess nobody really understands what I'm talking about. Maybe this could spark some discussion in the maintainers group, I mean nothing really changes since there's the renovate bot that updates things whenever a newer version is available (the bot's configuration allows for ranges per package, etc, but the json have specific versions!!!)

**Comment by brianteeman** [comment]
/me l;eaves this to cleverer people than me. am in over my head now

**Comment by dgrammatiko** [comment]
> it seems to be a windows issue that needs different sytax when using node_env than you have used.

can you confirm that `set NODE_ENV=PRODUCTION ` works? (I can't test it, not on windows)

**Comment by brianteeman** [comment]
still no luck

**Comment by brianteeman** [comment]
just to confirm that npm install with 4.2 completes successfuly

File: .drone.yml (hunk 1 above)
Comment (obuisard) [reply .drone.yml]: Thanks @heelc29. I need someone to approve this, as I cannot approve changes on my own PR

**Comment by obuisard** [comment]
Thank you, Brian @brianteeman for testing this and @dgrammatiko for looking into it. 
I will try the changes you submitted and try it as well. I am on Windows but use WSL with Debian.

**Comment by dgrammatiko** [comment]
@obuisard could your try this:

```json
    "build:js": "node build/build.js --compile-js",
    "build:css": "node build/build.js --compile-css",
    "build:bs5": "node build/build.js --compile-bs",
    "build:com_media": "set NODE_ENV=PRODUCTION && node build/build.js --com-media",
    "build:com_media:dev": "set NODE_ENV=DEVELOPMENT && node build/build.js --com-media",
    "watch": "node build/build.js --watch",
    "watch:com_media": "node build/build.js --watch-com-media",
    "lint:js": "eslint --config build/.eslintrc --ignore-pattern '/media/' --ext .es6.js,.es6,.vue .",
    "lint:css": "stylelint --config build/.stylelintrc.json \"administrator/components/com_media/resources/**/*.scss\" \"administrator/templates/**/*.scss\" \"build/media_source/**/*.scss\" \"templates/**/*.scss\" \"installation/template/**/*.scss\"",
    "install": "node build/build.js --prepare",
    "update": "node build/build.js --copy-assets && node build/build.js --build-pages && node build/build.js --compile-js && node build/build.js --compile-css && node build/build.js --compile-bs && set NODE_ENV=PRODUCTION && node build/build.js --com-media",
    "gzip": "node build/build.js --gzip",
    "versioning": "node build/build.js --versioning",
    "browserlist:update": "npx browserslist@latest --update-db",
    "cypress:install": "cypress install",
    "cypress:open": "cypress open",
    "cypress:run": "cypress run"
```

**Comment by dgrammatiko** [comment]
@obuisard one more try, add this at the end of the package.json

```json
  "optionalDependencies": {
     "sass-embedded-linux-x64": "^1.57.1"
   }
```

In case it's the same issue I had here: https://github.com/joomla/joomla-cms/pull/39400

**Comment by obuisard** [comment]
Still no luck

**Comment by brianteeman** [comment]
tested again.

I think the problem occurs because the file media\com_media\js\media-manager.js is not being created first 

image

**Comment by dgrammatiko** [comment]
@brianteeman could you try changing this: https://github.com/joomla/joomla-cms/blob/1c2bb9ae6dfa0aec903226b4c979dba2c7b2d358/build/build-modules-js/javascript/build-com_media-js.es6.js#L122-L132

to:

```js
 if (isProduction) {
    // eslint-disable-next-line no-console
    console.log('✅ ES2017 Media Manager ready');
    minifyJs('media/com_media/js/media-manager.js');
    return buildLegacy(resolve('media/com_media/js/media-manager.js'));
  } else {
    // eslint-disable-next-line no-console
    console.log('✅ ES2017 Media Manager ready');
    copyFile('media/com_media/js/media-manager.js', 'media/com_media/js/media-manager.js');
    return '';
}
```

**Comment by brianteeman** [comment]
1. same error
2. the file you linked to was the version for 4.2 - the version in this branch is different

**Comment by brianteeman** [comment]
and as I said these lines cant be the problem because they are trying to minify a file that does not exist (yet)

**Comment by dgrammatiko** [comment]
@brianteeman good point, @obuisard please replace the contents of `build/build-modules-js/javascript/build-com_media-js.es6.js` with:

```js
const { resolve } = require('path');
const { copyFile } = require('fs').promises;
const rollup = require('rollup');
const { nodeResolve } = require('@rollup/plugin-node-resolve');
const replace = require('@rollup/plugin-replace');
const { babel } = require('@rollup/plugin-babel');
const VuePlugin = require('rollup-plugin-vue');
const commonjs = require('@rollup/plugin-commonjs');
const { minifyJs } = require('./minify.es6.js');
require('dotenv').config();

const inputJS = 'administrator/components/com_media/resources/scripts/mediamanager.es6.js';
const isProduction = process.env.NODE_ENV !== 'DEVELOPMENT';

const buildLegacy = async (file) => {
  // eslint-disable-next-line no-console
  console.log('Building Legacy Media Manager...');

  const bundle = await rollup.rollup({
    input: file,
    plugins: [
      nodeResolve(),
      commonjs(),
      babel({
        exclude: 'node_modules/core-js/**',
        babelHelpers: 'bundled',
        babelrc: false,
        presets: [
          [
            '@babel/preset-env',
            {
              corejs: '3.8',
              useBuiltIns: 'usage',
              targets: {
                ie: '11',
              },
              loose: true,
              bugfixes: true,
              modules: false,
              ignoreBrowserslistConfig: true,
            },
          ],
        ],
      }),
    ],
    external: [],
  });

  await bundle.write({
    format: 'iife',
    sourcemap: false,
    name: 'JoomlaMediaManager',
    file: 'media/com_media/js/media-manager-es5.js',
  });

  // closes the bundle
  await bundle.close();

  // eslint-disable-next-line no-console
  console.log('Legacy Media Manager ready ✅');

  minifyJs('media/com_media/js/media-manager-es5.js');
};

module.exports.mediaManager = async () => {
  // eslint-disable-next-line no-console
  console.log('Building Media Manager ES Module...');

  const bundle = await rollup.rollup({
    input: resolve(inputJS),
    plugins: [
      VuePlugin({
        target: 'browser',
        css: false,
        compileTemplate: true,
        template: {
          isProduction,
        },
      }),
      nodeResolve(),
      commonjs(),
      replace({
        'process.env.NODE_ENV': JSON.stringify((process.env.NODE_ENV && process.env.NODE_ENV.toLocaleLowerCase()) || 'production'),
        __VUE_OPTIONS_API__: true,
        __VUE_PROD_DEVTOOLS__: isProduction,
        preventAssignment: true,
      }),
      babel({
        exclude: 'node_modules/core-js/**',
        babelHelpers: 'bundled',
        babelrc: false,
        presets: [
          [
            '@babel/preset-env',
            {
              targets: {
                browsers: [
                  '> 1%',
                  'not ie 11',
                  'not op_mini all',
                ],
              },
              loose: true,
              bugfixes: false,
              ignoreBrowserslistConfig: true,
            },
          ],
        ],
      }),
    ],
  });

  await bundle.write({
    format: 'es',
    sourcemap: false,
    file: 'media/com_media/js/media-manager.js',
  });

  // closes the bundle
  await bundle.close();

  if (isProduction) {
    // eslint-disable-next-line no-console
    console.log('✅ ES2017 Media Manager ready');
    minifyJs('media/com_media/js/media-manager.js');
    return buildLegacy(resolve('media/com_media/js/media-manager.js'));
  }
  // eslint-disable-next-line no-console
  console.log('✅ ES2017 Media Manager ready');
  copyFile('media/com_media/js/media-manager.js', 'media/com_media/js/media-manager.js');
  return '';
};

module.exports.watchMediaManager = async () => {
  // eslint-disable-next-line no-console
  console.log('Watching Media Manager js+vue files...');
  // eslint-disable-next-line no-console
  console.log('=========');

  const watcher = rollup.watch({
    input: resolve(inputJS),
    plugins: [
      VuePlugin({
        target: 'browser',
        css: false,
        compileTemplate: true,
        template: {
          isProduction: true,
        },
      }),
      nodeResolve(),
      replace({
        preventAssignment: true,
        'process.env.NODE_ENV': JSON.stringify('production'),
      }),
      babel({
        exclude: 'node_modules/core-js/**',
        babelHelpers: 'bundled',
        babelrc: false,
        presets: [
          [
            '@babel/preset-env',
            {
              targets: {
                browsers: [
                  '> 1%',
                  'not ie 11',
                  'not op_mini all',
                ],
              },
              loose: true,
              bugfixes: false,
              ignoreBrowserslistConfig: true,
            },
          ],
        ],
      }),
    ],
    output: [
      {
        format: 'es',
        sourcemap: false,
        file: 'media/com_media/js/media-manager.js',
      },
      {
        format: 'es',
        sourcemap: false,
        file: 'media/com_media/js/media-manager.min.js',
      },
    ],
  });

  watcher.on('event', (event) => {
    if (event.code === 'BUNDLE_END') {
      // eslint-disable-next-line no-console
      console.log(`File ${event.output[0]} updated ✅
File ${event.output[1]} updated ✅
=========`);
    }
  });
};
```

**Comment by brianteeman** [comment]
tested with the replacement file

1. npm install  ✔️
2. npm run build:com_media ✔️
3. npm run build:com_media:dev✔️ (not sure what it does but no errors)

**Comment by dgrammatiko** [comment]
> (not sure what it does but no errors)

- it doesn't create the -es5.js...
- the minified file is not minified (it's a copy of the .js)

**Comment by obuisard** [comment]
Thanks Dimitris and Brian for trying to get to the bottom of this. Unfortunately, the build is still failing.

**Comment by dgrammatiko** [comment]
@obuisard it seems that 4.2 and 4.3 have different build tools, try replacing the contents with

<details>
<summary>build/build-modules-js/javascript/build-com_media-js.es6.js</summary>

```js
const { resolve } = require('path');
const { copyFile, writeFile } = require('fs').promises;
const rollup = require('rollup');
const { nodeResolve } = require('@rollup/plugin-node-resolve');
const replace = require('@rollup/plugin-replace');
const { babel } = require('@rollup/plugin-babel');
const VuePlugin = require('rollup-plugin-vue');
const commonjs = require('@rollup/plugin-commonjs');
const { minifyJsCode } = require('./minify.es6.js');

const inputJS = 'administrator/components/com_media/resources/scripts/mediamanager.es6.js';
const isProduction = process.env.NODE_ENV !== 'DEVELOPMENT';

const buildLegacy = async (file) => {
  // eslint-disable-next-line no-console
  console.log('Building Legacy Media Manager...');

  const bundle = await rollup.rollup({
    input: file,
    plugins: [
      nodeResolve(),
      commonjs(),
      babel({
        exclude: 'node_modules/core-js/**',
        babelHelpers: 'bundled',
        babelrc: false,
        presets: [
          [
            '@babel/preset-env',
            {
              corejs: '3.8',
              useBuiltIns: 'usage',
              targets: {
                ie: '11',
              },
              loose: true,
              bugfixes: true,
              modules: false,
              ignoreBrowserslistConfig: true,
            },
          ],
        ],
      }),
    ],
    external: [],
  });

  await bundle.write({
    format: 'iife',
    sourcemap: false,
    name: 'JoomlaMediaManager',
    file: 'media/com_media/js/media-manager-es5.js',
  })
    .then((value) => minifyJsCode(value.output[0].code))
    .then((content) => {
      // eslint-disable-next-line no-console
      console.log('✅ Legacy Media Manager ready');
      return writeFile(resolve('media/com_media/js/media-manager-es5.min.js'), content.code, { encoding: 'utf8', mode: 0o644 });
    })
    .catch((error) => {
      // eslint-disable-next-line no-console
      console.error(error);
    });

  // closes the bundle
  await bundle.close();
};

module.exports.mediaManager = async () => {
  // eslint-disable-next-line no-console
  console.log('Building Media Manager ES Module...');

  const bundle = await rollup.rollup({
    input: resolve(inputJS),
    plugins: [
      VuePlugin({
        target: 'browser',
        css: false,
        compileTemplate: true,
        template: {
          isProduction,
        },
      }),
      nodeResolve(),
      commonjs(),
      replace({
        'process.env.NODE_ENV': JSON.stringify((process.env.NODE_ENV && process.env.NODE_ENV.toLocaleLowerCase()) || 'production'),
        __VUE_OPTIONS_API__: true,
        __VUE_PROD_DEVTOOLS__: isProduction,
        preventAssignment: true,
      }),
      babel({
        exclude: 'node_modules/core-js/**',
        babelHelpers: 'bundled',
        babelrc: false,
        presets: [
          [
            '@babel/preset-env',
            {
              targets: {
                browsers: [
                  '> 1%',
                  'not ie 11',
                  'not op_mini all',
                ],
              },
              loose: true,
              bugfixes: false,
              ignoreBrowserslistConfig: true,
            },
          ],
        ],
      }),
    ],
  });

  bundle.write({
    format: 'es',
    sourcemap: !isProduction,
    file: 'media/com_media/js/media-manager.js',
  })
    .then((value) => minifyJsCode(value.output[0].code))
    .then((content) => {
      if (isProduction) {
        // eslint-disable-next-line no-console
        console.log('✅ ES2017 Media Manager ready');
        writeFile(resolve('media/com_media/js/media-manager.min.js'), content.code, { encoding: 'utf8', mode: 0o644 });
        return buildLegacy(resolve('media/com_media/js/media-manager.js'));
      } else {
        // eslint-disable-next-line no-console
        console.log('✅ ES2017 Media Manager ready');
        writeFile(resolve('media/com_media/js/media-manager.js'), content.code, { encoding: 'utf8', mode: 0o644 });
        return;
      }
    })
    .then(() => buildLegacy(resolve('media/com_media/js/media-manager.js')))
    .catch((error) => {
      // eslint-disable-next-line no-console
      console.error(error);
    });

  // closes the bundle
  await bundle.close();
};

module.exports.watchMediaManager = async () => {
  // eslint-disable-next-line no-console
  console.log('Watching Media Manager js+vue files...');
  // eslint-disable-next-line no-console
  console.log('=========');

  const watcher = rollup.watch({
    input: resolve(inputJS),
    plugins: [
      VuePlugin({
        target: 'browser',
        css: false,
        compileTemplate: true,
        template: {
          isProduction: true,
        },
      }),
      nodeResolve(),
      replace({
        preventAssignment: true,
        'process.env.NODE_ENV': JSON.stringify('production'),
      }),
      babel({
        exclude: 'node_modules/core-js/**',
        babelHelpers: 'bundled',
        babelrc: false,
        presets: [
          [
            '@babel/preset-env',
            {
              targets: {
                browsers: [
                  '> 1%',
                  'not ie 11',
                  'not op_mini all',
                ],
              },
              loose: true,
              bugfixes: false,
              ignoreBrowserslistConfig: true,
            },
          ],
        ],
      }),
    ],
    output: [
      {
        format: 'es',
        sourcemap: false,
        file: 'media/com_media/js/media-manager.js',
      },
      {
        format: 'es',
        sourcemap: false,
        file: 'media/com_media/js/media-manager.min.js',
      },
    ],
  });

  watcher.on('event', (event) => {
    if (event.code === 'BUNDLE_END') {
      // eslint-disable-next-line no-console
      console.log(`✅ File ${event.output[0]} updated
✅ File ${event.output[1]} updated
=========`);
    }
  });
};
```
</details>

**Comment by obuisard** [comment]
No matter what, we get:
```
Processing Vendor file: media/vendor/webcomponentsjs/js/webcomponents-bundle.js
Error: Can't call SyncProcess.yield() after the process has exited.
npm ERR! code 1
npm ERR! path /********/src
npm ERR! command failed
npm ERR! command sh -c -- node build/build.js --prepare
```

**Comment by dgrammatiko** [comment]
@obuisard change https://github.com/joomla/joomla-cms/blob/23c43ed546a2c024fb93b3c271d5eb5929502bb1/.drone.yml#L26 to  `node:16-bullseye-slim` and ask @HLeithner to sign the change so that the tests could run... (it took me quite some time to figure out what was wrong. What happens when someone debugs on a blank screen)

File: .drone.yml [hunk 2]
```diff
@@ -30,21 +22,9 @@ steps:
...
-  - name: phan
-    image: joomlaprojects/docker-images:php7.4-ast
-    depends_on: [ phpcs ]
-    failure: ignore
-    commands:
-      - ./libraries/vendor/bin/phan
-
   - name: npm
-    image: node:16-bullseye-slim
+    image: node:16-alpine
```
Comment (dgrammatiko) [inline .drone.yml]: Revert this!

**Comment by dgrammatiko** [comment]
@obuisard I think the .drone.yml is wrong, 4.3 should have cypress not codeception for the tests. I'll say just paste the one form 4.3 directly, eg https://raw.githubusercontent.com/joomla/joomla-cms/4.3-dev/.drone.yml

File: build/build-modules-js/javascript/build-com_media-js.es6.js [hunk 3]
```diff
@@ -112,15 +117,22 @@ module.exports.mediaManager = async () => {
...
-      // eslint-disable-next-line no-console
-      console.log('✅ ES2017 Media Manager ready');
-
-      return writeFile(resolve('media/com_media/js/media-manager.min.js'), content.code, { encoding: 'utf8', mode: 0o644 });
+      if (isProduction) {
+        // eslint-disable-next-line no-console
+        console.log('✅ ES2017 Media Manager ready');
+        writeFile(resolve('media/com_media/js/media-manager.min.js'), content.code, { encoding: 'utf8', mode: 0o644 });
+        return buildLegacy(resolve('media/com_media/js/media-manager.js'));
+      } else {
```
Comment (dgrammatiko) [inline build/build-modules-js/javascript/build-com_media-js.es6.js]:
```suggestion
      }
```

File: build/build-modules-js/javascript/build-com_media-js.es6.js [hunk 4]
```diff
@@ -112,15 +117,22 @@ module.exports.mediaManager = async () => {
...
+        // eslint-disable-next-line no-console
+        console.log('✅ ES2017 Media Manager ready');
+        writeFile(resolve('media/com_media/js/media-manager.js'), content.code, { encoding: 'utf8', mode: 0o644 });
+        return;
+      }
```
Comment (dgrammatiko) [inline build/build-modules-js/javascript/build-com_media-js.es6.js:130]:
```suggestion
```

File: build/build-modules-js/javascript/build-com_media-js.es6.js [hunk 5]
```diff
@@ -112,15 +117,22 @@ module.exports.mediaManager = async () => {
...
-      console.log('✅ ES2017 Media Manager ready');
-
-      return writeFile(resolve('media/com_media/js/media-manager.min.js'), content.code, { encoding: 'utf8', mode: 0o644 });
+      if (isProduction) {
+        // eslint-disable-next-line no-console
+        console.log('✅ ES2017 Media Manager ready');
+        writeFile(resolve('media/com_media/js/media-manager.min.js'), content.code, { encoding: 'utf8', mode: 0o644 });
+        return buildLegacy(resolve('media/com_media/js/media-manager.js'));
+      } else {
+        // eslint-disable-next-line no-console
```
Comment (dgrammatiko) [inline build/build-modules-js/javascript/build-com_media-js.es6.js]:
```suggestion
      // eslint-disable-next-line no-console
```

File: build/build-modules-js/javascript/build-com_media-js.es6.js [hunk 6]
```diff
@@ -112,15 +117,22 @@ module.exports.mediaManager = async () => {
...
+        console.log('✅ ES2017 Media Manager ready');
```
Comment (dgrammatiko) [inline build/build-modules-js/javascript/build-com_media-js.es6.js]:
```suggestion
      console.log('✅ ES2017 Media Manager ready');
```

File: build/build-modules-js/javascript/build-com_media-js.es6.js [hunk 7]
```diff
@@ -112,15 +117,22 @@ module.exports.mediaManager = async () => {
...
+        writeFile(resolve('media/com_media/js/media-manager.js'), content.code, { encoding: 'utf8', mode: 0o644 });
```
Comment (dgrammatiko) [inline build/build-modules-js/javascript/build-com_media-js.es6.js]:
```suggestion
      writeFile(resolve('media/com_media/js/media-manager.js'), content.code, { encoding: 'utf8', mode: 0o644 });
```

File: build/build-modules-js/javascript/build-com_media-js.es6.js [hunk 8]
```diff
@@ -112,15 +117,22 @@ module.exports.mediaManager = async () => {
...
+        return;
```
Comment (dgrammatiko) [inline build/build-modules-js/javascript/build-com_media-js.es6.js]:
```suggestion
```

File: build/build-modules-js/javascript/build-com_media-js.es6.js (hunk 3 above)
Comment (dgrammatiko) [reply build/build-modules-js/javascript/build-com_media-js.es6.js]: @obuisard can you merge these changes so the linter won't complain?

File: build/build-modules-js/javascript/build-com_media-js.es6.js (hunk 8 above)
Comment (obuisard) [reply build/build-modules-js/javascript/build-com_media-js.es6.js]: I don't understand this one

Comment (dgrammatiko) [reply build/build-modules-js/javascript/build-com_media-js.es6.js]: Just delete the `return;`

File: build/build-modules-js/javascript/build-com_media-js.es6.js (hunk 3 above)
Comment (dgrammatiko) [inline build/build-modules-js/javascript/build-com_media-js.es6.js]:
```suggestion
      }
```

File: build/build-modules-js/javascript/build-com_media-js.es6.js (hunk 4 above)
Comment (dgrammatiko) [reply build/build-modules-js/javascript/build-com_media-js.es6.js:130]: also this needs merging

File: build/build-modules-js/javascript/build-com_media-js.es6.js [hunk 9]
```diff
@@ -112,15 +117,20 @@ module.exports.mediaManager = async () => {
...
+        // eslint-disable-next-line no-console
+        console.log('✅ ES2017 Media Manager ready');
+        writeFile(resolve('media/com_media/js/media-manager.min.js'), content.code, { encoding: 'utf8', mode: 0o644 });
+        return buildLegacy(resolve('media/com_media/js/media-manager.js'));
+      }
       // eslint-disable-next-line no-console
       console.log('✅ ES2017 Media Manager ready');
-
-      return writeFile(resolve('media/com_media/js/media-manager.min.js'), content.code, { encoding: 'utf8', mode: 0o644 });
+      writeFile(resolve('media/com_media/js/media-manager.js'), content.code, { encoding: 'utf8', mode: 0o644 });
```
Comment (dgrammatiko) [inline build/build-modules-js/javascript/build-com_media-js.es6.js]:
```suggestion
      return writeFile(resolve('media/com_media/js/media-manager.js'), content.code, { encoding: 'utf8', mode: 0o644 });
```

**Comment by obuisard** [comment]
THANK YOU, Dimitris for helping sort this out. Thanks to Brian for also testing things out and all for your advice.

**Pull request merged**